btree/btree.c

```diff
@@ -3263,33 +3272,33 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
 		return P_LOCK;
 	}
 	case P_LOCK:
-		if (!locked(tree))
+		if (!lock_acquired)
 			return lock_op_init(&bop->bo_op, &bop->bo_i->i_nop,
 					    bop->bo_arbor->t_desc, P_CHECK);
 		else
 			return P_CHECK;
```

Comment:
We can just fall down in instead of `else P_CHECK`.

Reply:
made changes, Also made respective changes in btree-conf. Please check.